runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -1530,7 +1542,92 @@ J9::CodeGenerator::lowerTreeIfNeeded(
 
    }
 
+void
+J9::CodeGenerator::rerunLivenessAnalysis()
+   {
+   auto comp = self()->comp();
+   auto trHeapMemory = self()->trHeapMemory();
+   auto trMemory = self()->trMemory();
+
+   if (!comp->getFlowGraph()->getStructure())
+      {
+      comp->getFlowGraph()->setStructure(TR_RegionAnalysis::getRegions(comp));
+      }
+
+      {
+      TR_BitVector *liveVars = NULL;
+      int32_t numLocals = 0;
+      TR::AutomaticSymbol *a;
+      ListIterator<TR::AutomaticSymbol> locals(&comp->getMethodSymbol()->getAutomaticList());
+      for (a = locals.getFirst(); a != NULL; a = locals.getNext())
+         {
+         // Mark collected locals as initialized. We will reset this property for
+         // any locals that are live at the start of the method.
+         //
+         if (a->isCollectedReference() &&
+            (!comp->getOption(TR_MimicInterpreterFrameShape) ||
+             !comp->areSlotsSharedByRefAndNonRef() ||
+             a->isSlotSharedByRefAndNonRef()))
+               a->setInitializedReference();
+         ++numLocals;
+         }
+
+      if (comp->getOption(TR_EnableAggressiveLiveness))
+         {
+         TR::ParameterSymbol *p;
+         ListIterator<TR::ParameterSymbol> parms(&comp->getMethodSymbol()->getParameterList());
+         for (p = parms.getFirst(); p != NULL; p = parms.getNext())
+            ++numLocals;
+         }
+
+      if (numLocals > 0 && (performTransformation(comp, "Rerunning liveness for CodeGen numLocals %d\n", numLocals)))
```

Comment:
The rerun shouldn't be optional. It must happen or we'll have functional issue. Could you use `dumpOptDetails` inside the if statement?